### PR TITLE
Fix:10071 - Accept watchOptions.aggregateTimeout be 0

### PR DESCRIPTION
**Pull request by jeffin143**
**What kind of change does this PR introduce?**
Bugfix : #10071 

Credits : @simonhowey

**Did you add tests for your changes?**

No
**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

File: lib/Watching.js [hunk 1]
```diff
@@ -43,7 +43,9 @@ class Watching {
 			this.watchOptions = {};
 		}
 		this.watchOptions.aggregateTimeout =
-			this.watchOptions.aggregateTimeout || 200;
+			this.watchOptions.aggregateTimeout === 0
+				? 0
+				: this.watchOptions.aggregateTimeout || 200;
```
Comment (evilebottnawi) [inline lib/Watching.js]: `if (this.watchOptions.aggregateTimeout !== undefined) ? this.watchOptions.aggregateTimeout : 200`

Comment (jeffin143) [reply lib/Watching.js]: @evilebottnawi you meant - 

```
this.watchOptions.aggregateTimeout = 
       (this.watchOptions.aggregateTimeout !== undefined ) ? 
                 this.watchOptions.aggregateTimeout : 200 ;
```

Comment (evilebottnawi) [reply lib/Watching.js]: yes

**Comment by simonhowey** [comment]
Thanks Jeffin!

FWIW I think:

https://github.com/webpack/watchpack/blob/c0221fc3be199b34711909c6c1245fc1054c7a51/lib/watchpack.js#L67

needs to be changed as well for the 0 to make it through to where it gets used. Could be wrong - have not looked at it recently.

File: lib/Watching.js (hunk 1 above)
Comment (sokra) [reply lib/Watching.js]:
``` js
if (typeof this.watchOptions.aggregateTimeout !== "number") {
  this.watchOptions.aggregateTimeout = 200;
}
```

Comment (jeffin143) [reply lib/Watching.js]: Done :)

**Comment by sokra** [comment]
@simonhowey seem to be right. Please also send a PR to watchpack.

**Comment by jeffin143** [comment]
@sokra , friendly - ping

**Comment by sokra** [comment]
Thanks

You missed the upgrade of the watchpack dep in webpack, but I did that in another PR.

**Pull request merged**